conans/test/functional/generators/cmake_find_package_multi_test.py

```diff
@@ -526,3 +526,121 @@ def build(self):
         self.assertIn("component libs: $<$<CONFIG:Debug>:;>;$<$<CONFIG:MinSizeRel>:;>;"
                       "$<$<CONFIG:RelWithDebInfo>:;>;$<$<CONFIG:Release>:system_lib_component;",
                       t.out)
+
+
+@pytest.mark.tool_cmake
+class TestNoNamespaceTarget:
+    """ This test case uses build-modules feature to create a target without a namespace. This
+        target uses targets create by Conan (build_modules are included after Conan targets)
+    """
+
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile, CMake
+
+        class Recipe(ConanFile):
+            settings = "os", "compiler", "arch"
+            exports_sources = ["src/*", "build-module.cmake"]
+            generators = "cmake"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure(source_folder="src")
+                cmake.build()
+
+            def package(self):
+                self.copy("*.h", dst="include", src="src")
+                self.copy("*.lib", dst="lib", keep_path=False)
+                self.copy("*.dll", dst="bin", keep_path=False)
+                self.copy("*.dylib*", dst="lib", keep_path=False)
+                self.copy("*.so", dst="lib", keep_path=False)
+                self.copy("*.a", dst="lib", keep_path=False)
+                self.copy("build-module.cmake", dst="share/cmake")
+
+            def package_info(self):
+                self.cpp_info.libs = ["library"]
+                module = os.path.join("share", "cmake", "build-module.cmake")
+                self.cpp_info.build_modules = module
+    """)
+
+    build_module = textwrap.dedent("""
+        message(">> Build-module is included")
+
+        if(NOT TARGET nonamespace)
+            add_library(nonamespace INTERFACE IMPORTED)
+            target_link_libraries(nonamespace INTERFACE library::library)
+        endif()
```

Comment:
What happen if a consumer relies on another generator than cmake_find_package(_multi) (cmake for example)? library::library doesn't exist anymore, so I guess target_link_libraries(nonamespace INTERFACE library::library) will fail?

Maybe this?

```suggestion
        if(NOT TARGET nonamespace AND TARGET library::library)
            add_library(nonamespace INTERFACE IMPORTED)
            target_link_libraries(nonamespace INTERFACE library::library)
        endif()
```

Reply:
I wrote a comment, but it went away with some of the commits 🤷 .

As you can see here, I can specify a different build-module per generator (https://github.com/conan-io/conan/pull/8338/commits/f2163c6c30037c0476aa48733a58e060bda65b82), otherwise it is the recipe author the one that needs to implement the required logic inside the build-module, of course.

Reply:
ok, so I understand that we could already (conan 1.32.0) provide non namespaced CMake imported targets in CCI recipe? conan 1.32.0 doesn't allow build_modules per generator but with some tricks it should work.

Reply:
Limitation in 1.32 is that `build_modules` are included before the Conan targets. It was changed here: https://github.com/conan-io/conan/pull/8130

Reply:
Surprinsingly, it works in 1.32: https://github.com/conan-io/conan-center-index/pull/4258

Reply:
Dammit! 😅   It should be some kind of (evil) CMake laziness resolution...

Reply:
So the idea is to create cmake build modules that are aliases to the conan generated ones?
Sounds good to me! 💯 
My next question is to add a build helper:

```python
from conan.tools.cmake import cmake_tools

cmake_tools.add_aliases(self.cpp_info.build_modules, {
    "library": "library::library",
    "library2": "library::library2",
})
```

Reply:
Please add a helper :heart:

Reply:
> Dammit! 😅 It should be some kind of (evil) CMake laziness resolution...

Actually, it seems to work in conan 1.32.0 if there are no components. It fails in opencv recipe: https://github.com/conan-io/conan-center-index/pull/4267

Reply:
About the helper, something for a feature request, not for this PR. Here we only wanted to check that adding targets without namespaces was possible.